pkg/util/docker/dockerfile/instructions.go

```diff
@@ -0,0 +1,68 @@
+package dockerfile
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/builder/command"
+)
+
+// A KeyValue can be used to build ordered lists of key-value pairs.
+type KeyValue struct {
```

Comment:
Why do you have this both in here and in pkg/build/builder/common.go? Also would it make sense to use kapi.EnvVar instead?